client/settings-recommendations/recommendations-eligibility-wrapper.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useUser, OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+const SHOW_MARKETPLACE_SUGGESTION_OPTION =
+	'woocommerce_show_marketplace_suggestions';
+
+const RecommendationsEligibilityWrapper: React.FC = ( { children } ) => {
+	const { currentUserCan } = useUser();
+
+	const isMarketplaceSuggestionsEnabled = useSelect( ( select ) => {
+		const { getOption, isResolving } = select( OPTIONS_STORE_NAME );
+
+		const isRequestingOptions = isResolving( 'getOption', [
```

Comment:
There has been some discussion around using `hasFinishedResolution` instead of `isResolving` to avoid a potential race condition.  Might be worth updating here.

Reply:
Thanks for the suggestion, done in 523a861 👍 
Out of curiosity, do you have a link to this discussion? I'd like to learn more about it :)

Reply:
I can't seem to find the conversation right now (cc @joelclimbsthings).  I believe in short it's due to `isResolving` returning `false` the first go around since the selector hasn't been called.

It returns `true` after the initial call, but this creates an opportunity where the option appears to have finished resolving even though resolution has not started yet.

Reply:
Ah, I see - I found this on that regard: https://github.com/Automattic/wp-calypso/pull/50327#issuecomment-784282630

Thanks for the explanation, TIL!

Reply:
Bit belated, but a little added context: [Sam's P2 post](p90Yrv-1Yq-p2).